components.html

```diff
@@ -0,0 +1,1381 @@
+
+
+
+      <!--COMPONENT LABEL-->
+      <p class="component-label">Dropdown</p>
+      <!--COMPONENT LABEL-->
+
+
+
+      <!--Dropdown-->
+      <select>
+        <option value="">store.test1.lewis.brains.us</option>
+        <option value="">store.test2.lewis.brains.us</option>
+        <option value="">store.test3.lewis.brains.us</option>
+      </select>
+
+
+
+      <!--COMPONENT LABEL-->
+      <p class="component-label">Tooltip Overlays</p>
+      <!--COMPONENT LABEL-->
+
+
+
+      <!--Speech bubble (arrow left)-->
+      <div class="flex-container">
+        <div class="speech-shadow">
+          <div class="speech-bubble-left">
+            <p>Text goes here</p>
+          </div>
+        </div>
+      </div>
+
+
+
+      <br />
+
+
+
+      <!--Speech bubble (arrow top)-->
+      <div class="flex-container">
+        <div class="speech-shadow">
+          <div class="speech-bubble-top">
+            <p>Text goes here</p>
+          </div>
+        </div>
+      </div>
+
+
+
+      <br />
+
+
+
+      <!--Speech bubble (button tooltip)-->
+      <div class="flex-container">
+        <div class="speech-shadow">
+          <div class="speech-bubble-left">
+            <p class="margin-bottom">Are you sure?</p>
+
+            <!--Icon button (Toolip)-->
+            <div class="flex-container">
+              <button class="btn btn-tooltip">
+                <div class="flex-container">
+                  <div>
+                    <svg class="icon-16 btn-tooltip-icon">
+                      <use xlink:href="img/icons.svg#check-white" />
+                    </svg>
+
+                  </div>
+                  Save
+              </button>
+
+              <!--Icon button (Toolip outline)-->
+              <button class="btn-tooltip-stroke">
+                Cancel
+              </button>
+            </div>
+          </div>
+        </div>
+      </div>
+
+
+
+      <br />
+
+
+
+      <!--Saved tooltip-->
+      <div class="flex-container">
+        <div class="tooltip-saved flex-container">
+          <svg class="icon-24 tooltip-saved-icon">
+            <use xlink:href="img/icons.svg#check-circular" />
+          </svg>
+          <p>Saved</p>
+        </div>
+      </div>
+
+
+
+      <!--COMPONENT LABEL-->
+      <p class="component-label">Search and Replace</p>
+      <!--COMPONENT LABEL-->
+
+
+
+      <!--Search and Replace-->
+      <table>
+        <tr class="flex-container table-heading bg-trans">
+          <td style="flex-grow:10;">
+            <h5>Search</h5>
+          </td>
+          <td style="flex-grow:2.3;">
+          </td>
+          <td style="flex-grow:10;">
+            <h5>Replace</h5>
+          </td>
+          <td>
+          </td>
+        </tr>
+
+        <tr class="flex-container">
+          <td style="flex-grow:10;">
+            <input type="checkbox" class="checkbox-default" name="" value="">
+            <label>File path goes here</label>
+          </td>
+          <td style="flex-grow:2;">
```

Comment:
Is there a reason for the inline styles here?

Reply:
I did some inline styles (flex-grow properties) in various places as there are quite a few columns in some of the layouts and I didn't know if it was worth settings classes for so many columns, particularly when some columns are just spacings.

@bradt what do you think?

Reply:
I would say that inline styles make sense only when the JS is going to be altering it dynamically. For example, in a progress bar, the JS might manipulate the width. Otherwise it should be a class so that if additional styles need to be applied to that element in the future it can be done relatively easily by editing the stylesheet.

Reply:
All fixed up now and commited.